geopandas/io/arrow.py

```diff
@@ -2,32 +2,39 @@
 import json
 import warnings
 
-from pandas import DataFrame
+from pandas import DataFrame, Series
 
 from geopandas._compat import import_optional_dependency
 from geopandas.array import from_wkb
 from geopandas import GeoDataFrame
 import geopandas
 from .file import _expand_user
 
-METADATA_VERSION = "0.1.0"
-# reference: https://github.com/geopandas/geo-arrow-spec
+METADATA_VERSION = "0.4.0"
+# reference: https://github.com/opengeospatial/geoparquet
 
 # Metadata structure:
 # {
 #     "geo": {
 #         "columns": {
 #             "<name>": {
-#                 "crs": "<WKT or None: REQUIRED>",
 #                 "encoding": "WKB"
+#                 "geometry_type": <str or list of str: REQUIRED>
+#                 "crs": "<PROJJSON or None: OPTIONAL>",
+#                 "orientation": "<'counterclockwise' or None: OPTIONAL>"
+#                 "edges": "planar"
+#                 "bbox": <list of [xmin, ymin, xmax, ymax]: OPTIONAL>
+#                 "epoch": <float: OPTIONAL>
 #             }
 #         },
+#         "primary_column": "<str: REQUIRED>",
+#         "version": "<METADATA_VERSION>",
+#
+#         # Additional GeoPandas specific metadata (not in metadata spec)
```

Comment:
Question: I'm assuming we want to continue writing GeoPandas specific metadata not in the GeoParquet spec, but this is a good time to consider removing that if we want to do so.

Reply:
Is there any benefit of keeping those? With the formalised spec, I don't see why not to remove it.

Reply:
One potential benefit would be that this bit of metadata tells us that a file was written by GeoPandas, in case there are later reports of issues in the wild and need to trace it back to something that originated here rather than elsewhere.  But given the simplicity of the encoding and spec at this point, I'm not sure that there is a lot of benefit to doing so; we'd likely need to do the same troubleshooting regardless.

Reply:
I would maybe leave it in here for now (although we might not have a specific use case right now, it also doesn't hurt to keep it?)

Reply:
I'd also be +1 on adding (optional?) `creator` metadata to the geoparquet spec. Maybe worth a discussion in the repo?

Reply:
@kylebarron sounds good. The only time it has come up in the geoparquet repo is this comment from Chris: https://github.com/opengeospatial/geoparquet/pull/4#issue-1030796247 (first bullet point).